pkg/util/hostname/providers.go

```diff
@@ -150,7 +181,14 @@ func GetWithProvider(ctx context.Context) (Data, error) {
 	var hostname string
 	var providerName string
 
-	for _, p := range providerCatalog {
+	var notUseIMDSv2 bool
+	if keyCache == legacyResolutionHostnameDataKey {
+		notUseIMDSv2 = true
+	} else {
+		notUseIMDSv2 = false
+	}
```

Comment:
I added a boolean to the getHostname method previously but wanted to avoid to change constructors to much, but you are right let's go back to the boolean it will be safer